src/sidebar/test/search-client-test.js

```diff
@@ -113,16 +114,15 @@ describe('SearchClient', () => {
     assert.calledWith(onResultCount, RESULTS.length);
   });
 
-  it('emits "end" only once', done => {
+  it('emits "end" only once', async () => {
     const client = new SearchClient(fakeSearchFn, { getPageSize: () => 2 });
     client.on('results', sinon.stub());
-    let emitEndCounter = 0;
-    client.on('end', () => {
-      emitEndCounter += 1;
-      assert.equal(emitEndCounter, 1);
-      done();
-    });
+    const { promise, resolve } = promiseWithResolvers();
+    client.on('end', () => resolve(1));
```

Comment:
Just kept the assertion over the value `1`, but this could actually be anything.